### ISSUE TITLE
Themosis integration issue

### ISSUE THREAD

**Issue by callumthomson** [jgrossi/corcel#388]
- Corcel Version: `v2.6.1`
- Framework Name & Version: `Themosis 1.3.2`
- PHP Version:  `7.2.0`
- Database Driver & Version: `mysql  Ver 14.14`

### Description:
Corcel Fails immediately when I try to retrieve data from the database using the `Post` model.

### Steps To Reproduce:

1. Include corcel in your themosis project
2. In your themosis theme, create a custom route and placing the following code:

For example
```php
Route::get('/random', function() {
    return Corcel\Model\Post::find(1);
});
```

You should find that you get the following error:
```
Call to undefined method Themosis\Foundation\Application::version()
```

The error is thrown on line 22 of jgrossi/corcel/src/Corcel.php

Thanks

**Comment by jgrossi** [comment]
Hey, thanks for asking. Themosis is made using Laravel's bases, so it seems it's breaking on the `app()` function. Can you what is the `app()` function returns an instance of `Themosis\Foundation\Application`, please?

**Comment by callumthomson** [comment]
Hi thanks for your swift response,

I replaced the code in my example with 

```php
Route::get('/random', function() {
    dd(app());
});
```
and the 2nd line of the output was
```
object(Themosis\Foundation\Application)[370]
```

**Comment by jgrossi** [comment]
Yep, got it. I would have to change Corcel's code to allow that. Gonna do that ASAP. Thanks for reporting.

**Comment by jgrossi** [comment]
@callumthomson is this fixed? I don't remember if I did that already! can you check please?

**Comment by callumthomson** [comment]
@jgrossi Line 22 of Corcel.php still reads as

```php
strpos(app()->version(), 'Lumen') === 0
```

so i guess not?

**Comment by arielatdev** [comment]
Hello, same issue here.
UP!

**Comment by jgrossi** [comment]
@callumthomson @Larsouille can you reinstall Corcel using the `2.6-fix-themosis` branch and confirm if it's solved?

```
"jgrossi/corcel": "dev-2.6-fix-themosis"
```

**Comment by jgrossi** [comment]
actually, Themosis uses almost all Laravel components, so I had to change the check to hard check if `app()` is an instance of Laravel or Lumen implementations, due to Themosis also uses the Laravel `Container` class. let me know if it needs any update, please 😎

**Comment by arielatdev** [comment]
Hello @jgrossi 
Nope, doesn't work, here's the stacktrace from my coworker:

ReflectionException: Class config does not exist in file /var/www/html/admin.site.frlocalhost/vendor/illuminate/container/Container.php on line 767
Stack trace:
  1. ReflectionException-&gt;() /var/www/html/admin.site.frlocalhost/vendor/illuminate/container/Container.php:767
  2. ReflectionClass-&gt;__construct() /var/www/html/admin.site.frlocalhost/vendor/illuminate/container/Container.php:767
  3. Illuminate\Container\Container-&gt;build() /var/www/html/admin.site.frlocalhost/vendor/illuminate/container/Container.php:646
  4. Illuminate\Container\Container-&gt;resolve() /var/www/html/admin.site.frlocalhost/vendor/illuminate/container/Container.php:601
  5. Illuminate\Container\Container-&gt;make() /var/www/html/admin.site.frlocalhost/vendor/jgrossi/corcel/src/App.php:30
  6. Corcel\App-&gt;instance() /var/www/html/admin.site.frlocalhost/vendor/jgrossi/corcel/src/App.php:40
  7. Corcel\App-&gt;config() /var/www/html/admin.site.frlocalhost/vendor/jgrossi/corcel/src/Model.php:182
  8. Corcel\Model-&gt;getConnectionName() /var/www/html/admin.site.frlocalhost/vendor/illuminate/database/Eloquent/Model.php:1167
  9. Illuminate\Database\Eloquent\Model-&gt;getConnection() /var/www/html/admin.site.frlocalhost/vendor/illuminate/database/Eloquent/Model.php:997
 10. Illuminate\Database\Eloquent\Model-&gt;newBaseQueryBuilder() /var/www/html/admin.site.frlocalhost/vendor/illuminate/database/Eloquent/Model.php:912
 11. Illuminate\Database\Eloquent\Model-&gt;newModelQuery() /var/www/html/admin.site.frlocalhost/vendor/illuminate/database/Eloquent/Model.php:950
 12. Illuminate\Database\Eloquent\Model-&gt;newQueryWithoutScopes() /var/www/html/admin.site.frlocalhost/vendor/illuminate/database/Eloquent/Model.php:901
 13. Illuminate\Database\Eloquent\Model-&gt;newQuery() /var/www/html/admin.site.frlocalhost/vendor/illuminate/database/Eloquent/Model.php:1570
 14. Illuminate\Database\Eloquent\Model-&gt;__call() /var/www/html/admin.site.frlocalhost/vendor/illuminate/database/Eloquent/Model.php:1582
 15. Illuminate\Database\Eloquent\Model-&gt;__callStatic() /var/www/html/admin.site.frlocalhost/vendor/jgrossi/corcel/src/Model/Option.php:81
 16. Corcel\Model\Option-&gt;get() /var/www/html/admin.site.frlocalhost/htdocs/content/plugins/wp-site-ecommerce-dashboard/resources/providers/DashboardCatalogsSettingsService.php:29
 17. site\Ecommerce\Dashboard\Services\DashboardCatalogsSettingsService-&gt;site\Ecommerce\Dashboard\Services\{closure}() /var/www/html/admin.site.frlocalhost/htdocs/cms/wp-includes/class-wp-hook.php:286
 18. WP_Hook-&gt;apply_filters() /var/www/html/admin.site.frlocalhost/htdocs/cms/wp-includes/class-wp-hook.php:310
 19. WP_Hook-&gt;do_action() /var/www/html/admin.site.frlocalhost/htdocs/cms/wp-includes/plugin.php:453
 20. do_action() /var/www/html/admin.site.frlocalhost/htdocs/cms/wp-admin/admin-ajax.php:99


Regards :)

**Comment by h-collector** [comment]
@Larsouille  for `'config'` to be resolved you can use another package like [wp-kit/config](https://github.com/wp-kit/config) otherwise Themosis only define `config.factory` (not really helpful) and it is not really compatible with most Laravel Packages.

**Comment by arielatdev** [comment]
@h-collector I'll check this out tomorrow, thanks for the tip! :)

**Comment by jgrossi** [comment]
@h-collector good call 👏actually the hard part is that Themosis uses a lot of Laravel features/packages, but Corcel works by default with Laravel:

1. if the app is Laravel you don't have to set database stuff. then Corcel only reads the `database.connection` config and call it.
2. if it's not Laravel it does not read that, but then you're forced to configure `illuminate/database`'s Capsule to make sure it's working.
3. when using Themosis that's the hard part. maybe I should find somehow to avoid getting the Laravel's config stuff and then getting database stuff using `WP_DATABASE` and stuff like that.

**Comment by jgrossi** [comment]
that would be something just for Themosis, actually! because it's Laravel but it's not 😂

**Comment by arielatdev** [comment]
Hello @jgrossi @h-collector , so wp-kit would not work I think.
Because of composer dependancies loop/issues:
wp-kit needs symfony/finder: 3.*
dev-2.6fix-themosis needs Illuminate/filesystem: 5.6*
Illuminate/filesytem needs symfony/finder: ~4.0

Cheers! :)

**Comment by h-collector** [comment]
You can try to make an alias for finder version, tried and works! :D
```
"require": {
    "php": ">=7.0.1",
    "johnpbloch/wordpress-core-installer": "^1.0",
    "johnpbloch/wordpress-core": "^4.9.5",
    "themosis/framework": "^1.3.3",
    "vlucas/phpdotenv": "^2.4",
    "filp/whoops": "^2.1",
    "guzzlehttp/guzzle": "^6.3",
    "illuminate/database": "^5.6",
    "illuminate/validation": "^5.6",
    "illuminate/pagination": "^5.6",
    "illuminate/filesystem": "^5.6",
    "illuminate/cache": "^5.6",
    "league/flysystem": "^1.0",
    "wp-kit/config": "^2.0",
    "symfony/finder": "4.1.1 as 3.0",
    "jgrossi/corcel" : "dev-2.6-fix-themosis",
    "symfony/var-dumper": "^4.0",
  }
```

**Comment by jgrossi** [comment]
@h-collector but this should be done at the Themosis level, not Corcel's, right?

**Comment by h-collector** [comment]
@jgrossi Yes, this is composer.json require section from my current project (without plugins) .

I updated from Corcel v2.5.7 (alongside illuminate packages 5.5 to 5.6), just needed to also add new config file (corcel), looks like it works (dev-2.6-fix-themosis I mean) .

**Comment by h-collector** [comment]
Btw. there could be also some checks `->bound('config')` with defaults.
Themosis by default use global Eloquent without config but it should change with new 1.4 version
(I'm using Illuminate DatabaseServiceProvider with config)

> This package allows you to use WordPress as backend (admin panel) and retrieve its data using 
Eloquent, with any PHP project or even framework.

Not `any`-more :D

**Comment by ninthspace** [comment]
Related to this, I just happen to have a website that's unrelated to Laravel, that has an app() function (which returns an instance of a class). Correspondingly Corcel::isLaravel fails because that doesn't instance doesn't have a version() method. (Actually, nevermind, I'll update to Corcel 2.7 from 2.5 and see if this issue goes away)

**Comment by ninthspace** [comment]
Ah, 2.7 doesn't have the dev-2.6-fix-themosis changes. So I installed that instead and my issue was fixed. 👍
